Python/optimizer.c

```diff
@@ -1669,12 +1669,15 @@ executor_clear(PyObject *op)
      * free the executor unless we hold a strong reference to it
      */
     _PyExecutorObject *cold = _PyExecutor_GetColdExecutor();
+    _PyExecutorObject *cold_dynamic = _PyExecutor_GetColdDynamicExecutor();
     Py_INCREF(executor);
     for (uint32_t i = 0; i < executor->exit_count; i++) {
         executor->exits[i].temperature = initial_unreachable_backoff_counter();
         _PyExecutorObject *e = executor->exits[i].executor;
-        executor->exits[i].executor = cold;
-        Py_DECREF(e);
+        executor->exits[i].executor = NULL;
+        if (e != cold && e != cold_dynamic) {
+            executor_clear((PyObject *)e);
```

Comment:
```suggestion
            Py_DECREF(e);
```

Using `executor_clear` could clear a live executor.

Reply:
`Py_DECREF` doesn't work

> The problem is that executors must be forcefully cleared upon invalidation, otherwise their reference cycles prevent clearing of invalid executors and those remain in the bytecode.

Reply:
Also `executor_clear` will never clear a live executor --- `executor_clear` only, unlinks, detaches, and decrefs an executor, it doesn't call `PyObject_GC_DEL` on it.